config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -339,4 +348,18 @@ public void setEntityId(Long entityId) {
     public void setEntityType(String entityType) {
         this.entityType = entityType;
     }
+
+    @Override
+    public ConfigOrigin getOrigin() {
+        return origin;
+    }
+
+    public boolean isRemote()
+    {
+        return origin != null && !origin.isLocal();
```

Comment:
> So you mean we could have something like
> ```
> private ConfigOrigin origin = new NullOrigin();
> ```

Yes, thats what I meant. Since the initialisation would be at the field level, magical loader will not make it null, unless there is an explicit call to make it null or change the value. 

>Do you want a commit added in this PR or should we refactor with your changes too?

Fine with either. If you do not want to make the change now, its fine. I will make the change in my PR if you are ok with it.